pkg/router/template/router.go

```diff
@@ -490,51 +521,38 @@ func (r *templateRouter) AddRoute(id string, route *routeapi.Route, host string)
 	r.lock.Lock()
 	defer r.lock.Unlock()
 
-	frontend, _ := r.findMatchingServiceUnit(id)
+	frontend, _ := r.findMatchingServiceUnit(serviceID)
 
 	//create or replace
-	frontend.ServiceAliasConfigs[backendKey] = config
-	r.state[id] = frontend
-	r.cleanUpdates(id, backendKey)
+	config.ServiceUnitNames[frontend.Name] = weight
+	r.state[backendKey] = config
+	r.serviceUnits[serviceID] = frontend
+	//r.cleanUpdates(serviceID, backendKey)
 	return true
 }
 
-// cleanUpdates ensures the route is only under a single service key.  Backends are keyed
```

Comment:
You answered that before, sorry for asking again.

On Thu, Jun 9, 2016 at 3:45 PM, Rajat Chopra <notifications@github.com>
wrote:

> In pkg/router/template/router.go
> <https://github.com/openshift/origin/pull/9119#discussion_r66508731>:
>
> >  	return true
> >  }
> >
> > -// cleanUpdates ensures the route is only under a single service key.  Backends are keyed
>
> No need for the cleanup now.. we have got two tables where the object is
> referenced. So we do not need to delete the duplicate objects. Takes care
> of the situation where a route is created before a service is created too.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/openshift/origin/pull/9119/files/21555064205e47d56f8a15f3d4d838ad0ba51002#r66508731>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABG_p31Bf8FKPiffp4pbhyK8TGojK_tKks5qKG1igaJpZM4Ir8cY>
> .
>